### ISSUE TITLE
Feature Request: Option to insert script at bottom of HTML body

### ISSUE THREAD

**Issue by iquiw** [onozaty/redmine-view-customize#70]
I'd like to insert View Customize script that manages Gantt chart at bottom of HTML body.

Gant chart HTML looks like the following.

```html
<html>
...
// (A) view customize scripts
...
<script src=".../gantt.js"/>
...
// main contents
...
<script>
$(function () {
// gantt chart initialize
});
</script>
...
// (B) 
</body>
</html>
```

It is difficult to manage generated gantt chart at `(A)`, so I'd like to insert script at `(B)`.

**Comment by onozaty** [comment]
Thank you for your feature request.

What you want to do may be possible without additional features.
With the `window` load event, you'd be able to run it after the Gantt chart script.

```
$(window).on('load', function() {
  // your scripts
});
```

This is because the `window` load event occurs after `$(function{})`.

Would you please give it a try?
If you can't get around it here, we'd like to add a feature.

**Comment by iquiw** [comment]
Thanks!

I've tried it, but no success.
Actually, it worked with RedMica v1.1, but not works with RedMica v1.2.

I will try again and paste the script here later.

Anyway, I am trying to implement it...
https://github.com/iquiw/redmine-view-customize/tree/body-bottom

**Comment by onozaty** [comment]
Thanks for trying.

If the load event doesn't work, it may not work even if you use View customize to handle it.

**Comment by iquiw** [comment]
Here is the script, which saves and restores Gantt chart subject column width.

```javascript
// https://stackoverflow.com/a/2523758
$.widget("ui.resizable", $.ui.resizable, {
  resizeTo: function(newSize) {
    let start = new $.Event("mousedown", { pageX: 0, pageY: 0 });
    this._mouseStart(start);
    this.axis = 'se';
    let end = new $.Event("mouseup", {
      pageX: newSize.width - this.originalSize.width,
      pageY: newSize.height - this.originalSize.height
    });
    this._mouseDrag(end);
    this._mouseStop(end);
  }
});

$(window).on('load', function() {
//$(function() {
  const GANTT_SUBJECT_WIDTH_KEY = 'redmine-gantt-subject-width';
  let $subject = $('.gantt_subjects_column.ui-resizable');
  $subject.resizable({
    stop: function(event, ui) {
      localStorage.setItem(GANTT_SUBJECT_WIDTH_KEY, ui.size.width);
    }
  });
  let width = localStorage.getItem(GANTT_SUBJECT_WIDTH_KEY);
  if (width) {
    $subject.resizable('resizeTo', { width: width, height: $subject.height() });
  }
});
```

As far as I tried the following combinations with RedMica v1.2,

* Browser: Chrome/Firefox
* Position: `Head of all pages`/`Bottom of all pages`
* Load: `$(window).on('load'`/`$(function()`

With Chrome, it works in the following cases.
|                       | `Head of all pages` | `Bottom of all pages` |
| ---                   | ---                 | ---                   |
| `$(window).on('load'` | O                   | O                     |
| `$(function()`        | X                   | O                     |

With Firefox, it works only when using `$(function()` and `Bottom of all pages`.
|                       | `Head of all pages` | `Bottom of all pages` |
| ---                   | ---                 | ---                   |
| `$(window).on('load'` | X (*1)              | X                     |
| `$(function()`        | X                   | O                     |

(*1) worked with RedMica v1.1.

**Comment by onozaty** [comment]
Thank you for the information.

I don't understand why the window load doesn't work well in Firefox.
Is there some kind of error?

Either way, we are thinking of including this feature.

**Comment by iquiw** [comment]
> I don't understand why the window load doesn't work well in Firefox.

It could be a bug of Firefox though I am not sure.

> Is there some kind of error?

No error, `$('.gantt_subjects_column.ui-resizable').length` is 0 in the case.

> Either way, we are thinking of including this feature.

Thanks!
I will try to add unit test and update README.
I am afraid I cannot update Chinese translation.

**Comment by onozaty** [comment]
> > Either way, we are thinking of including this feature.
> 
> Thanks!
> I will try to add unit test and update README.
> I am afraid I cannot update Chinese translation.

It is okay to have a pull request in the current state.
I'll take care of the rest.

Thanks.

**Comment by iquiw** [comment]
> It is okay to have a pull request in the current state.

Submitted PR.

> I'll take care of the rest.

Thank you so much!!

**Comment by onozaty** [comment]
Released at v2.8.0.

Thanks!
